dcc-submission-core/src/main/java/org/icgc/dcc/submission/release/model/ReleaseView.java

```diff
@@ -63,34 +57,10 @@ public ReleaseView(Release release, List<Project> projects, Map<String, List<Sub
     this.releaseDate = release.releaseDate;
     this.dictionaryVersion = release.dictionaryVersion;
 
-    for (val project : projects) {
-      String projectKey = project.getKey();
-      val optional = release.getSubmission(projectKey);
-      checkState(optional.isPresent(), "Could not find project '%s' in release '%s'", projectKey, name);
-
-      val submission = optional.get();
-      DetailedSubmission detailedSubmission = new DetailedSubmission(submission, project);
-      detailedSubmission.setSubmissionFiles(submissionFiles.get(projectKey));
-      this.submissions.add(detailedSubmission);
-
-      Integer stateCount = this.summary.get(detailedSubmission.getState());
-      if (stateCount == null) {
-        stateCount = 0;
-      }
-      stateCount++;
-      this.summary.put(detailedSubmission.getState(), stateCount);
+    for (val submissionState : submissionsStates) {
```

Comment:
http://stackoverflow.com/questions/25441088/group-by-counting-in-java8-stream-api:

```java
val summary  = submissionsStates.stream().collect(groupingBy(identity(), counting()));
```